core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -361,14 +346,13 @@ sealed abstract class MultiSigSigner
     val signers = signersWithPubKeys.map(_.signFunction)
 
     val requiredSigs = spendingInfoToSatisfy.scriptPubKey.requiredSigs
+    val component =
+      if (spendingInfo.isInstanceOf[P2SHSpendingInfo])
+        sigComponent(spendingInfoToSatisfy, unsignedTx)
```

Comment:
I'm worried here about the case of `P2SH(Conditional(MultiSig, ...))` and things like it. Won't it be wrong to use `spendingInfoToSatisfy` as you're supposed to sign `Conditional(MultiSig, ...)` not `MultiSig`?